src/main/java/com/getindata/connectors/http/sink/HttpSink.java

```diff
@@ -0,0 +1,78 @@
+package com.getindata.connectors.http.sink;
+
+import org.apache.flink.connector.base.sink.AsyncSinkBase;
+import org.apache.flink.connector.base.sink.writer.BufferedRequestState;
+import org.apache.flink.connector.base.sink.writer.ElementConverter;
+import org.apache.flink.core.io.SimpleVersionedSerializer;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+
+public class HttpSink<InputT> extends AsyncSinkBase<InputT, HttpSinkRequestEntry> {
+  private final String endpointUrl;
+
+  protected HttpSink(
+      ElementConverter<InputT, HttpSinkRequestEntry> elementConverter,
+      int maxBatchSize,
+      int maxInFlightRequests,
+      int maxBufferedRequests,
+      long maxBatchSizeInBytes,
+      long maxTimeInBufferMS,
+      long maxRecordSizeInBytes,
+      String endpointUrl
+  ) {
+    super(elementConverter, maxBatchSize, maxInFlightRequests, maxBufferedRequests, maxBatchSizeInBytes,
+          maxTimeInBufferMS, maxRecordSizeInBytes
+    );
+    this.endpointUrl =
+        Preconditions.checkNotNull(endpointUrl, "The endpoint URL must not be null when initializing HTTP Sink.");
+    Preconditions.checkArgument(!endpointUrl.isEmpty(), "The endpoint URL must be set when initializing HTTP Sink.");
```

Comment:
You have validation in builder for all arguments except those two, for which you placed validation here, why?
BTW what about dnpointUrl = "    " :) 

You could use `org.apache.flink.util.StringUtils::isNullOrWhitespaceOnly`